src/accelerate/inference.py

```diff
@@ -0,0 +1,98 @@
+import math
+from types import MethodType
+from typing import Literal
+
+from pippy.IR import Pipe, PipeSplitWrapper, annotate_split_points
+from pippy.PipelineStage import PipelineStage
+
+from .state import PartialState
+from .utils import (
+    calculate_maximum_sizes,
+    convert_bytes,
+    infer_auto_device_map,
+    send_to_device,
+)
+
+
+ParallelMode = Literal["sequential", "pipeline_parallel"]
+
+
+def generate_device_map(model, num_processes: int = 1, no_split_module_classes=None):
+    """
+    Calculates the device map for `model` with an offset for PiPPy
+    """
+    if num_processes == 1:
+        return infer_auto_device_map(model, no_split_module_classes=no_split_module_classes, clean_result=False)
+    model_size, shared = calculate_maximum_sizes(model)
+
+    # Split into `n` chunks for each GPU
+    memory = (model_size + shared[0]) / num_processes
+    memory = convert_bytes(memory)
+    value, ending = memory.split(" ")
+
+    # Add a chunk to deal with potential extra shared memory instances
+    memory = math.ceil(float(value)) * 1.1
+    memory = f"{memory} {ending}"
+    device_map = infer_auto_device_map(
+        model,
+        max_memory={i: memory for i in range(num_processes)},
+        no_split_module_classes=no_split_module_classes,
+        clean_result=False,
+    )
+    return device_map
```

Comment:
Let's add the `max_memory` arg, so that the user can set it by himself if we want it. So if `max_memory` is not None and `num_processes != 1`, we would change its value with the above allocation. Then call `infer_auto_device_map` once the end of the function.